Add automated working group reporting system for TOC quarterly reviews

rkoster opened the pull request:
## Summary

• Implements comprehensive automation system for quarterly working group activity reports to enhance TOC governance and oversight
• Separates data extraction from AI analysis using OpenCode Run integration for strategic insights and consistent reporting

## Key Changes

- **Data Extraction Script** (`scripts/extract_wg_activity.py`): Pure GitHub API data extraction for commits, PRs, issues, releases, and RFCs
- **OpenCode Integration** (`scripts/generate_working_group_update.py`): Leverages OpenCode Run for AI-powered analysis of raw activity data
- **Documentation**: Comprehensive automation docs in `toc/working-groups/updates/README.md` with usage examples
- **Example Report**: Sample quarterly report for Foundational Infrastructure working group
- **Infrastructure**: Added `tmp/` directory for temporary data files with proper git ignore

## Benefits

- **Standardized Reporting**: Consistent quarterly reports across all working groups
- **Reduced Manual Effort**: Automated data collection and analysis
- **Strategic Insights**: AI-powered analysis identifies trends, health indicators, and recommendations
- **Data-Driven Governance**: Enables TOC to make informed decisions based on comprehensive activity data

The system maintains clean separation between data extraction (GitHub API) and analysis (OpenCode Run), making it maintainable and extensible for future enhancements.

FI Working group update generated by the new automation can be found [here](https://github.com/cloudfoundry/community/blob/fiwg-update/toc/working-groups/updates/2025-09-02-foundational-infrastructure.md).

rkoster committed:
Add automated working group reporting system for TOC quarterly reviews

rkoster committed:
Add automatic previous update detection for incremental reporting

- Script now checks for existing update files in toc/working-groups/updates/
- Uses previous update date as start date when no date is specified
- Falls back to 3 months lookback if no previous update exists
- Enables incremental reporting without manual date specification

rkoster committed:
Highlight IPv6 dual-stack as primary strategic initiative in FI working group update

- Add RFC-0038 IPv6 dual-stack support as the leading major initiative
- Emphasize the historic significance of this networking evolution
- Include comprehensive technical details and community impact
- Update activity metrics and summary to reflect IPv6 importance
- Add IPv6 implementation opportunities in forward-looking section
- Ensure the most important track of work is prominently featured

rkoster committed:
Refocus FI working group update on IPv6 implementation and UAA modernization

- Replace AliCloud emphasis with UAA identity management modernization
- Highlight active IPv6 dual-stack implementation progress (BOSH core, AWS CPI, testing)
- Feature UAA ExternalLoginAuthenticationManager refactoring by @adrianhoelzl-sap
- Add Storage CLI AWS SDK v2 migration initiative
- Update activity metrics to reflect actual development priorities
- Based on comprehensive data dump analysis revealing key missing work areas

rkoster committed:
Limit major initiative summaries to 2 paragraphs for conciseness

- Condensed all major strategic initiatives to exactly 2 paragraphs each
- Maintained technical depth while improving readability and focus
- Updated generation script to enforce 2-paragraph constraint for future reports
- Preserved all key technical details and related work links
- Enhanced report accessibility for TOC consumption

rkoster committed:
Fix duplicate IPv6 header in working group update

- Remove duplicate 'IPv6 Dual-Stack Implementation in Active Development' header
- Clean up formatting inconsistency in major initiatives section

rkoster committed:
Improve working group report format and generation

- Add proper issue linking format (descriptive link + org/repo#number shorthand)
- Remove speculative 'Looking Forward' section to prevent hallucination
- Update generation script to use existing reports as templates
- Ensure consistent formatting across all working group updates

rkoster committed:
Streamline report structure with inline contributor recognition

- Move contributor recognition inline within initiative descriptions
- Add GitHub profile links for all contributors using [Name](https://github.com/username) format
- Remove separate Community Contributors Recognition section
- Consolidate summary to single paragraph with key contributors highlighted
- Update generation script to enforce inline recognition with GitHub links

rkoster committed:
Integrate PR/issue links directly into descriptive text

- Remove all separate 'Related Work' sections from initiatives
- Embed PR and issue links inline within the paragraph text that describes the work
- Maintain dual link format (descriptive link + org/repo#number shorthand)
- Update generation script to require inline integration and prohibit separate link sections
- Create natural text flow with links embedded where work is described

rkoster committed:
Enforce 40-word paragraph limit for improved readability

- Add maximum 40 words per paragraph requirement to generation script
- Break down all long paragraphs in FI report to meet 40-word limit
- Split summary from 1 long paragraph (83 words) to 2 concise paragraphs (24, 35 words)
- Condense all initiative sections to 2 paragraphs of maximum 40 words each
- Maintain technical accuracy and comprehensive PR/issue linking while improving readability

rkoster committed:
Improve PR linking format and eliminate duplication

- Remove shorthand org/repo#number format from all PR links
- Use descriptive link text that explains the work performed instead of repo references
- Add rule to prevent duplicate PR links within the same paragraph
- Update generation script to prefer descriptive links over repo#PR format
- Maintain comprehensive linking while improving readability and avoiding redundancy

rkoster committed:
Streamline report structure by eliminating redundant sections

- Remove Community Impact Areas and Activity Breakdown sections that provide no value
- Integrate RFC developments directly into relevant Major Strategic Initiatives
- Add new Shared Infrastructure and CI/CD Modernization initiative for RFC-0041
- Embed RFC-0038 into IPv6 section and RFC-0043 into Storage CLI section
- Update generation script to integrate RFCs within initiatives rather than separate section
- Create cohesive structure showing how governance enables technical implementation

rkoster committed:
Remove redundant Summary section and Major Strategic Initiatives heading

- Delete Summary section that provided redundant high-level overview
- Remove 'Major Strategic Initiatives' heading while keeping all initiative content
- Promote all initiative subsections from h3 to h2 level for cleaner hierarchy
- Update generation script to eliminate Summary section requirement
- Create direct, focused structure that jumps immediately into technical work
- Maintain all technical content while eliminating introductory redundancy

rkoster committed:
Prioritize RFC-related activity in working group reports

rkoster committed:
Implement streamlined working group report format with RFC prioritization

rkoster committed:
Fix frontmatter positioning in working group report

rkoster commented:
In the light of: https://github.com/cloudfoundry/community/pull/1297, I might have used a little bit of AI :grimacing:

rkoster committed:
Clean-up Readme

aramprice commented:
Would be helpful to add a `requirements.txt`, perhaps under `scripts/` or elsewhere.

aramprice commented:
Seeing the following error:
```
  File "~/workspace/community/./scripts/extract_wg_activity.py", line 719, in <module>
    main()
    ~~~~^^
  File "~/workspace/community/./scripts/extract_wg_activity.py", line 712, in main
    with open(args.output, 'w') as f:
         ~~~~^^^^^^^^^^^^^^^^^^
FileNotFoundError: [Errno 2] No such file or directory: 'tmp/foundational-infrastructure_activity.json'
```

Code should ensure `tmp/` exists, or `tmp/.gitkeep` should be created / added.

aramprice committed:
Add generated FIWG update

title: "Foundational Infrastructure Working Group Update"
date: 2026-06-22
period: September 2, 2025 - June 22, 2026
